### ISSUE TITLE
Typescript not reported about extra key in returned object in function

### ISSUE THREAD

**Issue by ctr52** [Microsoft/TypeScript#25178]
**TypeScript Version:**  2.9

**Search Terms:** excess key in returned object in function
**Code**

```ts
interface ITestFunctionReturns {
    key1: number,
    key2: string,
}
type ITestFunction = () => ITestFunctionReturns

const testFunction: ITestFunction = () => ({
    asd: true,
    key1: 123,
    key2: 'str',
})
```

**Expected behavior:** typescript will report about extra key in the object

**Actual behavior:** I must point the interface in yet another place

```ts
interface ITestFunctionReturns {
    key1: number,
    key2: string,
}
type ITestFunction = () => ITestFunctionReturns

const testFunction: ITestFunction = (): ITestFunctionReturns => ({
    asd: true,
    key1: 123,
    key2: 'str',
})
```

**Comment by binbinfrog** [comment]
I‘ve also encountered this problem, this is what my code looks like

```
interface ITest {
  [name: string]: () => { one1: 1 };
}

const TestImpl: ITest = {
  one: function() {
    return { one1: 1, one2: 2 };
  }
};
```

**Comment by RyanCavanaugh** [comment]
Duplicate #12632
